Assets/MixedRealityToolkit.Extensions/Webrtc/Scripts/WebrtcPeerVideo.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using UnityEngine;
+using Microsoft.MixedReality.Toolkit.Extensions.Webrtc.Marshalling;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Webrtc
+{
+    /// <summary>
+    /// Provides video frame hooks for Webrtc
+    /// </summary>
+    public class WebrtcPeerVideo : MonoBehaviour
+    {
+        /// <summary>
+        /// Instance of a local video player which will have frame data written to it
+        /// </summary>
+		[Tooltip("Video player instance that local stream data will be written to")]
+        public WebRtcVideoPlayer LocalPlayer;
+
+        /// <summary>
+        /// Instance of a remote video player which will have frame data written to it
+        /// </summary>
+		[Tooltip("Video player instance that remote stream data will be written to")]
+        public WebRtcVideoPlayer RemotePlayer;
+
+        /// <summary>
+        /// The underlying peer
+        /// </summary>
+        private IPeer peer = null;
+
+        /// <summary>
+        /// Internal queue used for storing and processing local video frames
+        /// </summary>
+        private FrameQueue localFrameQueue = new FrameQueue(3);
+
+        /// <summary>
+        /// Internal queue used for storing and processing remote video frames
+        /// </summary>
+        private FrameQueue remoteFrameQueue = new FrameQueue(5);
+
+        /// <summary>
+        /// Initialization handler
+        /// 
+        /// TODO(bengreenier): This could be auto-bound with editor script
+        /// </summary>
```

Comment:
nit: maybe do a code search for your directory to make sure all todos are removed